homeassistant/components/aprilaire/sensor.py

```diff
@@ -0,0 +1,260 @@
+        device_class=SensorDeviceClass.ENUM,
+        options=list(set(DEHUMIDIFICATION_STATUS_MAP.values())),
+        status_key=Attribute.DEHUMIDIFICATION_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1],
+        value_key=Attribute.DEHUMIDIFICATION_STATUS,
+        value_fn=lambda _, value: DEHUMIDIFICATION_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="humidification_status",
+        translation_key="humidification_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=list(set(HUMIDIFICATION_STATUS_MAP.values())),
+        status_key=Attribute.HUMIDIFICATION_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1, 2],
+        value_key=Attribute.HUMIDIFICATION_STATUS,
+        value_fn=lambda _, value: HUMIDIFICATION_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="ventilation_status",
+        translation_key="ventilation_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=list(set(VENTILATION_STATUS_MAP.values())),
+        status_key=Attribute.VENTILATION_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1],
+        value_key=Attribute.VENTILATION_STATUS,
+        value_fn=lambda _, value: VENTILATION_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="air_cleaning_status",
+        translation_key="air_cleaning_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=list(set(AIR_CLEANING_STATUS_MAP.values())),
+        status_key=Attribute.AIR_CLEANING_AVAILABLE,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[1],
+        value_key=Attribute.AIR_CLEANING_STATUS,
+        value_fn=lambda _, value: AIR_CLEANING_STATUS_MAP.get(value),
+    ),
+    AprilaireSensorDescription(
+        key="fan_status",
+        translation_key="fan_status",
+        device_class=SensorDeviceClass.ENUM,
+        options=list(set(FAN_STATUS_MAP.values())),
+        status_key=None,
+        status_sensor_available_value=None,
+        status_sensor_exists_values=[],
+        value_key=Attribute.FAN_STATUS,
+        value_fn=lambda _, value: FAN_STATUS_MAP.get(value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Aprilaire sensor devices."""
+
+    coordinator: AprilaireCoordinator = hass.data[DOMAIN][config_entry.unique_id]
+
+    assert config_entry.unique_id is not None
+
+    async_add_entities(
+        AprilaireSensor(coordinator, description, config_entry.unique_id)
+        for description in SENSOR_TYPES
+        if description.status_key is None
+        or description.status_sensor_exists_values is None
+        or coordinator.data.get(description.status_key)
+        in description.status_sensor_exists_values
```

Comment:
I think I asked before on the last PR, why did we do this again? Like the whole status thingy

Reply:
@joostlek It is getting all of the possible sensor types and including them if either:

1. `status_key` or `status_sensor_exists_values` is `None`, indicating that the sensor should always be added
2. The coordinator data for the `status_key` is one of the allowed values in `status_sensor_exists_values`

The reason for the slightly clunky logic with the `status_sensor_exists_values` is that the thermostat returns a value for the status for which multiple may correspond to the sensor being available. For instance, the Indoor temperature controlling sensor should show in the UI if the `INDOOR_TEMPERATURE_CONTROLLING_SENSOR_STATUS` value is either 0 (normal) or 1/2 (installed but in an error state).

Reply:
What do you think about maybe splitting the classes into the different sensors you are providing here? Even tho this works, you have a shared entity (And entity description) that doesn't have that much overlap.

For the controlling sensors `status_sensor_available_value`, `status_sensor_exists_values` and `value_fn` are all the same, while for the status ones, `status_sensor_available_value` are all the same.

Also since the `available` property is based around the difference in both sensors, would it make sense to maybe split the `AprilaireSensor` into 2 classes? (You could add a third to have 1 class inheriting both `BaseAprilaireEntity, SensorEntity` and then have the two other classes inherit this superclass). I think this way, the code is more clearer on what it does and when it does something and is easier for extending in the future.

But let me know what you think (don't see this as a must, just throwing an idea)

Reply:
@joostlek Personally I think that would introduce more complexity than it's worth and it would reduce the flexibility, especially considering future uses that are possible. In terms of the cases where the values are the same, the `value_fn` won't be the same once we reintroduce the temperature conversion that I removed. Also, if this were to be extended to have more of the sensors supported, then the `status_sensor_available_value` and `status_sensor_exists_values` wouldn't be the same, requiring refactoring at that point.

I'm _open_ to changing it up, but I don't see a great way of doing it that wouldn't just add a lot of bloat for no real benefit.

Reply:
But keep in mind you can then introduce the conversion in the abstraction for example. So you put the unit conversion in the `native_value` function of that sensor.

Do you already know what you want to add in future sensors? Maybe I can give it a shot.

(Feel free to message me on discord if you want to have some chats about it)

Reply:
@joostlek I have made some changes, I think it addresses what you're thinking. Let me know if you're meaning anything else.